site/en/docs/extensions/migrating/to-service-workers/index.md

```diff
@@ -109,10 +109,17 @@ Communicate between offscreen documents and extension service workers using [mes
 
 ## Convert localStorage to another type {: #convert-localstorage }
 
-The web platform's [`Storage`](https://developer.mozilla.org/docs/Web/API/Storage) interface (accessible from `window.localStorage`) cannot be used in a service worker. To address this do one of the following:
+The web platform's [`Storage`](https://developer.mozilla.org/docs/Web/API/Storage) interface (accessible from `window.localStorage`) cannot be used in a service worker. To address this do one of two things. First, you can replace it with calls to another storage mechanism. The [`chrome.storage.local`](/docs/extensions/reference/storage/#property-local) namespace will serve most use cases, but [other options](/docs/extensions/mv3/service_workers/service-worker-lifecycle/#persist-data) are available.
 
-* Move its calls to an [offscreen document](/docs/extensions/reference/offscreen/).
-* Replace it with calls to another storage mechanism. The [`chrome.storage.local`](/docs/extensions/reference/storage/#property-local) namespace will serve most use cases. [Other options](/docs/extensions/mv3/service_workers/service-worker-lifecycle/#persist-data) are available.
+You can also move its calls to an [offscreen document](/docs/extensions/reference/offscreen/).
+
+1. Create an offscreen document with a conversion routine and a [`runtime.onMessage`](/docs/extensions/reference/runtime/#event-onMessage) handler.
```

Comment:
I think the lead-in sentence here makes it sound like we're going to give general advice, but then the steps are very much a migration routine. Could we either:

- Add a sentence before along the lines of "For example, to migrate data previously stored in localStorage to another mechanism:"
- Replace this with an actual long term way of doing storage, e.g starting an offscreen document, sending some data and then storing it. And then similar to read data.

Hope that makes sense :)

Reply:
Given that this is a migration page I think the former would be more useful, but I'm ok with either.

Reply:
Done.